lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js

```diff
@@ -301,6 +396,36 @@ describe('LMSFilePicker', () => {
     assert.calledWith(onSelectFile, file);
   });
 
+  it('does not invoke `onSelectFile` if chosen file is empty', async () => {
+    const onSelectFile = sinon.stub();
+    const wrapper = renderFilePicker({ onSelectFile });
+    await fakeApiCall;
+    wrapper.update();
+
+    const file = null;
+    wrapper.find('FileList').prop('onUseFile')(file);
+    assert.notCalled(onSelectFile);
+  });
+
+  it('updates the folder path when a user chooses a folder', async () => {
+    const onSelectFile = sinon.stub();
+    const wrapper = renderFilePicker({ onSelectFile, withBreadcrumbs: true });
+    await fakeApiCall;
+    wrapper.update();
+
+    const file = fakeFiles[1]; // This is a folder
+    wrapper.find('FileList').prop('onUseFile')(file);
+    // Folders cannot be selected as a file...
+    assert.notCalled(onSelectFile);
+
+    await fakeApiCall;
+    wrapper.update();
```

Comment:
This should be replaced with a suitable "wait until file list is not in a loading state" wait.